src/Text/Pandoc/Writers/Markua.hs

```diff
@@ -0,0 +1,501 @@
+    let marker = literal "[^" <> num' <> literal "]: "
+    return $ hang 4 marker (nestle contents)
+
+-- | Convert list of Pandoc block elements to Markua
+blockListToMarkua :: PandocMonad m
+                  => WriterOptions
+                  -> [Block]
+                  -> Markua m (Doc Text)
+blockListToMarkua opts blocks =
+  mconcat <$> mapM (blockToMarkua opts) blocks
+
+-- | Convert Pandoc block element to Markua
+blockToMarkua :: PandocMonad m
+              => WriterOptions
+              -> Block
+              -> Markua m (Doc Text)
+blockToMarkua opts (Plain inlines) = do
+    contents <- inlineListToMarkua opts inlines
+    return $ contents <> cr
+
+blockToMarkua opts (Para inlines) = do
+    contents <- inlineListToMarkua opts inlines
+    -- set explicit a blankline before and after para to be conform with Markua
+    return $ blankline <> contents <> blankline
+
+blockToMarkua opts (LineBlock inlines) = do -- TODO nowrap richtig?
+    contents <- mconcat <$> mapM (inlineListToMarkua opts) inlines
+    return $ nowrap contents <> blankline
+
+blockToMarkua _ (CodeBlock attr txt) = do
+    let attributes = attrToMarkua attr
+    let contents = literal txt
+    let shell = literal "```"
+    return $ blankline <> attributes <> cr <> shell <> cr <> contents <> cr 
+                       <> shell <> blankline
+
+-- no attr like format are printed
+blockToMarkua _ (RawBlock _ txt) = do
+    report $ NoTranslation "raw content is writen out in original form and" 
+    return $ cr <> literal txt <> cr 
+
+blockToMarkua opts (BlockQuote blks) = do
+    contents <- blockListToMarkua opts blks
+    let quoteHead = "> "
+    return $ prefixed quoteHead (nestle contents) <> blankline
+
+blockToMarkua opts ((OrderedList (start,style,delim) items)) = do
+    let listAttributes = (start,style,delim)
+    let markers  = orderedListMarkers listAttributes
+    contents <- zipWithM (orderedListItemToMarkua opts) markers items
+    return $ cr <> mconcat contents <> blankline
+
+blockToMarkua opts (BulletList items) = do
+    contents <- mapM (bulletListItemToMarkua opts) items
+    return $ cr <> mconcat contents <> blankline
+    
+blockToMarkua opts (DefinitionList items) = do
+    contents <- mapM (definitionListItemToMarkua opts) items
+    return $ cr <> mconcat contents <> blankline
+
+-- every header gets automatically an attribute list
+blockToMarkua opts (Header level attr inlines) = do
+    contents <- inlineListToMarkua opts inlines
+    let attributes = attrToMarkua attr
+    let eqs = literal $ T.replicate level "#"
+    return $ blankline <> attributes <> cr 
+            <> eqs <> space <> contents <> blankline
+
+blockToMarkua _ HorizontalRule = return $ blankline <> "* * *" <> blankline
+
+-- div is also used to compute the bibliography
+-- if --citeproc and --bibliography is set (div with refs id is generated),
+-- then a bibliography header level 1 with a attr list above is inserted
+blockToMarkua opts (Div attr blks) = do
```

Comment:
Pattern matching might result in cleaner code. How about something like `blockToMarkua opts (Div ("refs",_,_) blks)` to handle the bibliography?